packages/common/src/services/local-storage/LocalStorage.ts

```diff
@@ -34,7 +35,7 @@ export class LocalStorage {
     return await this.localStorage.getItem(key)
   }
 
-  async getJSONValue(key: string) {
+  async getJSONValue<T>(key: string): Promise<T | null> {
```

Comment:
I like this, but also I've been thinking more about how this isn't "correct" from a static typing perspective. The true value is "unknown" as anything could be in there. (Similar to API requests). Part of me feels like using `as` here to coerce types makes it explicit that it's an unsafe thing, similar to a cast. The other part of me loves the handy type parameter. No need to change, just thought I'd share my musings.

Reply:
Very good point! The issue I have with `as` is that it's a bit of a broken window thing - imo ideally we'd have a codebase free of `as`, since it's usually a sign you're doing something wrong/taking a shortcut & having `as` as happy path feels a bit dangerous. But point taken that implicitly you're performing a dangerous coercion here, and maybe it's better to make it explicit. If it was Rust, we'd get around this with`dangerouslyGetJSONValue`. Runtime types wen

Reply:
Haha yeah, I mean on one hand that's the idea - forcing the use of `as` makes it scarier and hopefully makes the coder double check, but on the other hand totally agree that could lead to desensitizing to `as` anywhere